Source/items.cpp

```diff
@@ -1138,74 +1138,60 @@ void GetBookSpell(int i, int lvl)
 		item[i]._iCurs = ICURS_BOOK_GREY;
 }
 
-void GetStaffPower(int i, int lvl, int bs, unsigned char onlygood)
+void GetStaffPower(int i, int lvl, int bs, BOOL onlygood)
 {
-	int v4;         // esi
-	int v5;         // ebx
-	int v6;         // edx
-	int v7;         // ecx
-	int v9;         // edi
-	int v10;        // ecx
-	int v11;        // ST14_4
-	int v12;        // esi
-	char *v13;      // edi
-	int l[256];     // [esp+Ch] [ebp-484h]
-	char istr[128]; // [esp+40Ch] [ebp-84h]
-	int ia;         // [esp+48Ch] [ebp-4h]
-	char *v17;      // [esp+49Ch] [ebp+Ch]
-
-	v4 = lvl;
-	ia = i;
-	v5 = -1;
+	int l[256];
+	char istr[128];
+	int nl, j, preidx;
+	BYTE addok;
+
+	preidx = -1;
 	if (!random(15, 10) || onlygood) {

```

Comment:
`random(15, 10) == 0` maybe?

Reply:
I think we have neglected to do this for most of the random calls

Reply:
I think it's much more readable

Reply:
Yeah it aligns with our style guide so we probably should correct it

Reply:
I agree, the function `random` returns an `int` so should use == 0.